File: python/resistor-color-trio/resistor_color_trio.py
import re
mnemonic = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white"
]
def label(colors):
    num1 = mnemonic.index(colors[0])
    num2 = mnemonic.index(colors[1])
    zeros = int(mnemonic.index(colors[2]))
    out = str(num1) + str(num2)
    out = out.ljust(zeros + len(out), '0')
    num = int(out)
    tag = 'ohms'
    if num and num % 1000 == 0:
        num = num / 1000
        tag = 'kiloohms'
        if num % 1000 == 0:
            num = num / 1000
            tag = 'megaohms'
            if num % 1000 == 0:
                num = num / 1000
                tag = 'gigaohms'
    num = int(num)
    return f"{num} {tag}"

File: ruby/resistor-color-trio/resistor_color_trio.rb
class ResistorColorTrio
  ORDER = %w[black brown red orange yellow green blue violet grey white].freeze

  def initialize(colors)
    @colors = colors
  end

  def digit(index)
    ORDER.index(@colors[index])
  end

  def label
    value = [digit(0), digit(1)].join.to_i * 10 ** digit(2)

    units = value > 1000 ? "kiloohms" : "ohms"
    value /= 1000 if value > 1000

    return "Resistor value: #{value} #{units}"
  end
end
